core/commands/cmdenv/cidbase.go

```diff
@@ -0,0 +1,86 @@
+package cmdenv
+
+import (
+	path "gx/ipfs/QmQtg7N4XjAk2ZYpBjjv8B6gQprsRekabHBCnF6i46JYKJ/go-path"
+	cmds "gx/ipfs/Qma6uuSyjkecGhMFFLfzyJDPyoDtNJSHJNweDccZhaWkgU/go-ipfs-cmds"
+	cidenc "gx/ipfs/QmckgkstbdXagMTQ4e1DW2SzxGcjjudbqEvA5H2Rb7uvAT/go-cidutil/cidenc"
+	cmdkit "gx/ipfs/Qmde5VP1qUkyQXKCfmEUA7bP64V2HAptbJ7phuPp7jXWwg/go-ipfs-cmdkit"
+	mbase "gx/ipfs/QmekxXDhCxCJRNuzmHreuaT3BsuJcsjcXWNrtV9C8DRHtd/go-multibase"
+)
+
+var OptionCidBase = cmdkit.StringOption("cid-base", "Multi-base encoding used for version 1 CIDs in output.")
+var OptionOutputCidV1 = cmdkit.BoolOption("output-cidv1", "Upgrade CID version 0 to version 1 in output.")
```

Comment:
That would be very misleading.  This option only effects the Cid version that is outputted and nothing else, in particular in `ipfs add` it will not change the Hash.

Reply:
The reason for this option is because CIDv0 are not suppose to take a multihash so if use use `--cid-base=base32` CidV0 hashes will still output in base58btc which is not very helpful. So we decided that we (me @Stebalien daviddias whyrusleeping and few others) would automatically upgrade CIDv0 to CIDv1, but in the output only.

If `--cid-base` is given this is implied.  This was agreed upon between me and @Stebalien see https://github.com/ipfs/go-ipfs/pull/5289#issuecomment-415902770.

Reply:
My problem with this option is that it only allows to upgrade from 0 to 1, which can be a bit limiting.  
If this was an int (and still only affect output), this would also allow:
* Users to say they want to stick with CIDv0 after we switch to CIDv1 by default
* Us to reuse this option if we need it for another migration

Reply:
I can make this `--output-cid-version` but that will make this option more tedious to use and I don't see any benefit from it.  I really consider this option somewhat of a hack to handle the special case that `CIDv0` can not take a multibase prefix.

> Users to say they want to stick with CIDv0 after we switch to CIDv1 by default

`--output-cidv1=false` will do this

> Us to reuse this option if we need it for another migration

While theoretically true I am not sure if we will want to reuse this option.  This option is only present to deal with the special case that CIDv0 can not take a multibase prefix.  I doubt we will want to reuse it again.

Reply:
I admit that is we ever have a need to create a new CID version this option `--output-cidv1` would be misleading as it would imply that CIDv2 are downgraded to CIDv1 which was not the intention.  I originally used `--upgrade-cidv0` but changed it as as we might need that option to actually upgrade Cid's for example when pinning.  I am open to other names, but have a strong preference to something reasonable short and not something like  `--upgrade-cidv0-in-output`.

Reply:
What about something like `--force-multibase-cids` (or something like that). In the help text, we can explain that it auto-upgrades cidv0 to cidv1 in output. Unfortunately, this flag *doesn't* say what it actually does so it's not much better.

This is probably a distant non-issue but `--output-cidv1` could be a *bit* confusing if we ever get a cidv2.

(I don't have any strong opinions on this API)

Reply:
:+1: for `--force-multibase-cids`

Reply:
I am still in favor of `--ouuput-cidv1` because it is clear and to the point, although I do admit it could be confusing in a dissent future if we ever use CIDv2.  As an alternative how about `--force-cid-base`, this is a little clearer because it makes it obvious that this option is related to the `cid-base` option and because not everone will know what a `multibase` is.  The only thing I don't like about it is is not 100% accurate as this option will also convert to cidv1 for `base58btc` even though it is not strictly needed.

Other alternatives that I though of and dismissed, `--force-multibase-output` (makes it clear that the option effects the output), `--force-multibase` (shorter).

@Stebalien @magik6k @ipfs/go-team thoughts?

Reply:
We decided to go with `--upgrade-cidv0-in-output`.